tests/test_catalyst.py

```diff
@@ -0,0 +1,151 @@
+    ```
+    """
+
+    y = np.array(y, dtype='int')
+    input_shape = y.shape
+    if input_shape and input_shape[-1] == 1 and len(input_shape) > 1:
+        input_shape = tuple(input_shape[:-1])
+    y = y.ravel()
+    if not num_classes:
+        num_classes = np.max(y) + 1
+    n = y.shape[0]
+    categorical = np.zeros((n, num_classes), dtype=dtype)
+    categorical[np.arange(n), y] = 1
+    output_shape = input_shape + (num_classes,)
+    categorical = np.reshape(categorical, output_shape)
+    return categorical
+
+
+class Net(nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.conv1 = nn.Conv2d(1, 20, 5, 1)
+        self.conv2 = nn.Conv2d(20, 50, 5, 1)
+        self.fc1 = nn.Linear(4 * 4 * 50, 500)
+        self.fc2 = nn.Linear(500, 10)
+
+    def forward(self, x):
+        x = F.relu(self.conv1(x))
+        x = F.max_pool2d(x, 2, 2)
+        x = F.relu(self.conv2(x))
+        x = F.max_pool2d(x, 2, 2)
+        x = x.view(-1, 4 * 4 * 50)
+        x = F.relu(self.fc1(x))
+        x = self.fc2(x)
+        return x
+
+
+class TestCatalyst(unittest.TestCase):
+
+    def test_version(self):
+        self.assertIsNotNone(catalyst.__version__)
+    
+    def test_mnist(self):
+        utils.set_global_seed(42)
+        x_train = np.random.random((100, 1, 28, 28)).astype(np.float32)
+        y_train = _to_categorical(
+            np.random.randint(10, size=(100, 1)), 
+            num_classes=10
+        ).astype(np.float32)
+        x_valid = np.random.random((20, 1, 28, 28)).astype(np.float32)
+        y_valid = _to_categorical(
+            np.random.randint(10, size=(20, 1)), 
+            num_classes=10
+        ).astype(np.float32)
+
+        x_train, y_train, x_valid, y_valid = \
+            list(map(torch.tensor, [x_train, y_train, x_valid, y_valid]))
+
+        bs = 32
+        num_workers = 4
+        data_transform = transforms.ToTensor()
+
+        loaders = collections.OrderedDict()
+
+        trainset = torch.utils.data.TensorDataset(x_train, y_train)
+        trainloader = torch.utils.data.DataLoader(
+            trainset, batch_size=bs,
+            shuffle=True, num_workers=num_workers)
+
+        validset = torch.utils.data.TensorDataset(x_valid, y_valid)
+        validloader = torch.utils.data.DataLoader(
+            validset, batch_size=bs,
+            shuffle=False, num_workers=num_workers)
+
+        loaders["train"] = trainloader
+        loaders["valid"] = validloader
+
+        # experiment setup
+        num_epochs = 3
+        logdir = "./logs"
+
+        # model, criterion, optimizer
+        model = Net()
+        criterion = nn.BCEWithLogitsLoss()
+        optimizer = torch.optim.Adam(model.parameters())
+
+        # model runner
+        runner = SupervisedRunner()
+
+        # model training
+        runner.train(
+            model=model,
+            criterion=criterion,
+            optimizer=optimizer,
+            loaders=loaders,
+            logdir=logdir,
```

Comment:
Do you need to call some sort of `Close` method?

The tests shows a ResourceWarning:

```
unclosed file <_io.TextIOWrapper name='/working/logs/log.txt' mode='a' encoding='UTF-8'>
```